src/createTokenBridgePrepareTransactionReceipt.ts

```diff
@@ -0,0 +1,111 @@
+    );
+  }
+
+  return log;
+}
+
+function decodeOrbitTokenBridgeCreatedEventLog(log: Log<bigint, number>) {
+  const decodedEventLog = decodeEventLog({ ...log, abi: tokenBridgeCreator.abi });
+
+  if (decodedEventLog.eventName !== 'OrbitTokenBridgeCreated') {
+    throw new Error(
+      `Expected "OrbitTokenBridgeCreated" event but found: ${decodedEventLog.eventName}`,
+    );
+  }
+
+  return decodedEventLog;
+}
+
+type RedeemedRetryableTicket = {
+  status: L1ToL2MessageStatus.REDEEMED;
+  l2TxReceipt: EthersTransactionReceipt;
+};
+
+export type WaitForRetryablesParameters = {
+  orbitPublicClient: PublicClient;
+};
+
+export type WaitForRetryablesResult = [TransactionReceipt, TransactionReceipt];
+
+type GetTokenBridgeContractsParameters = {
+  parentChainPublicClient: PublicClient;
+};
+
+export type CreateTokenBridgeTransactionReceipt = TransactionReceipt & {
+  waitForRetryables(params: WaitForRetryablesParameters): Promise<void>;
+  getTokenBridgeContracts(): TokenBridgeContracts;
+};
+
+export function createTokenBridgePrepareTransactionReceipt(txReceipt: TransactionReceipt) {
+  return {
+    ...txReceipt,
+    waitForRetryables: async function ({
+      orbitPublicClient,
+    }: WaitForRetryablesParameters): Promise<WaitForRetryablesResult> {
+      const ethersTxReceipt = viemTransactionReceiptToEthersTransactionReceipt(txReceipt);
+      const l1TxReceipt = new L1TransactionReceipt(ethersTxReceipt);
+      const orbitProvider = publicClientToProvider(orbitPublicClient);
+      const messages = await l1TxReceipt.getL1ToL2Messages(orbitProvider);
+      const messagesResults = await Promise.all(messages.map((message) => message.waitForStatus()));
+
+      if (messagesResults.length !== 2) {
+        throw Error(`Unexpected number of retryable tickets: ${messagesResults.length}`);
+      }
+
+      if (messagesResults[0].status !== L1ToL2MessageStatus.REDEEMED) {
+        throw Error(`Unexpected status for retryable ticket: ${messages[0].retryableCreationId}`);
+      }
+
+      if (messagesResults[1].status !== L1ToL2MessageStatus.REDEEMED) {
+        throw Error(`Unexpected status for retryable ticket: ${messages[1].retryableCreationId}`);
+      }
+
+      return (
+        // these type casts are both fine as we already checked everything above
+        (messagesResults as unknown as [RedeemedRetryableTicket, RedeemedRetryableTicket])
+          //
+          .map((result) =>
+            ethersTransactionReceiptToViemTransactionReceipt(result.l2TxReceipt),
+          ) as WaitForRetryablesResult
+      );
+    },
+    getTokenBridgeContracts: async function ({
+      parentChainPublicClient,
+    }: GetTokenBridgeContractsParameters): Promise<TokenBridgeContracts> {
+      const eventLog = findOrbitTokenBridgeCreatedEventLog(txReceipt);
+      const decodedEventLog = decodeOrbitTokenBridgeCreatedEventLog(eventLog);
+      const { inbox } = decodedEventLog.args;
+
+      return await createTokenBridgeFetchTokenBridgeContracts({
```

Comment:
We have a smol situation here (for testing). Calling `createTokenBridgeFetchTokenBridgeContracts` will always return the TokenBridge contracts from the canonical TokenBridgeCreator, so when we create the TokenBridge using a newly deployed TokenBridgeCreator, we can't get the contracts here.

This makes sense anyway, because we want to enforce people using the canonical TokenBridgeCreator, but it's something to keep in mind for when we are testing.